api/v1alpha1/datadogagent_default.go

```diff
@@ -99,7 +102,10 @@ func IsDefaultedDatadogAgent(ad *DatadogAgent) bool {
 	}
 
 	if ad.Spec.Features != nil {
-		if !IsDefaultedOrchestratorExplorer(ad.Spec.Features.OrchestratorExplorer) {
+		if ad.Spec.Features.OrchestratorExplorer != nil && !IsDefaultedOrchestratorExplorer(ad.Spec.Features.OrchestratorExplorer) {
+			return false
+		}
+		if ad.Spec.Features.KubeStateMetricsCore != nil && !IsDefaultedKubeStateMetricsCore(ad.Spec.Features.KubeStateMetricsCore) {
```

Comment:
I think checking if the value is `nil` can be part of the function  `IsDefaultedKubeStateMetricsCore()`
```suggestion
		if ad.Spec.Features.KubeStateMetricsCore != nil && !IsDefaultedKubeStateMetricsCore(ad.Spec.Features.KubeStateMetricsCore) {
```

Reply:
in fact the function is already doing it :)

Reply:
true, but in this case I only want to check if it's defaulted if the main field is not nil.
with the above:
```
features:
    kubeStateMetricsCore:
         enabled: true
```
will not return `false`. Whereas if I don't split it, I would end up needing to mutate the orchestrator explorer field. Which I don't think we should do.